charms/reactive/relations.py

```diff
@@ -702,6 +702,7 @@ def remove_state(self, state):
         state = state.format(relation_name=self.relation_name)
         value = _get_flag_value(state)
         if not value:
+            clear_flag(state)
```

Comment:
If `value` is false-y then the state will be cleared; this is not what the PR is titled with, which suggests None.